### PR TITLE
feat(color): support analogous colors to prevent color conflict

### PR DESCRIPTION
**Pull request by stephenLYZ**
### SUMMARY

Currently we have a series of color schemes (aribnb, d3, superset...) Since we use recycle colors when number of items exceed available colors, the color conflict is easy to occur. For this case we use `tinycolor.analogous` API to get analogous colors to prevent color conflict and the color generation algorithm comes from color consistency.
### before
image

### after
image

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://github.com/apache/superset/issues/17443
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

cc @rusackas @geido

### CODE REVIEW

File: superset-frontend/packages/superset-ui-core/src/color/utils.ts [hunk 1]
```diff
@@ -51,3 +52,22 @@ export function getContrastingColor(color: string, thresholds = 186) {
 
   return r * 0.299 + g * 0.587 + b * 0.114 > thresholds ? '#000' : '#FFF';
 }
+
+export function getAnalogousColors(colors: string[], results: number) {
+  const generatedColors: string[] = [];
+  const ext = 3;
```
Comment (rusackas) [inline superset-frontend/packages/superset-ui-core/src/color/utils.ts:58]: I'm curious how we landed on 3 for this?

Comment (rusackas) [reply superset-frontend/packages/superset-ui-core/src/color/utils.ts:58]: A comment might be useful for future readers.

File: superset-frontend/packages/superset-ui-core/src/color/utils.ts [hunk 2]
```diff
@@ -51,3 +52,22 @@ export function getContrastingColor(color: string, thresholds = 186) {
...
+  const analogousColors = colors.map(color => {
+    const result = tinycolor(color).analogous(results + ext);
+    return result.slice(ext);
+  });
+
+  // [[A, AA, AAA], [B, BB, BBB]] => [A, B, AA, BB, AAA, BBB]
```
Comment (rusackas) [inline superset-frontend/packages/superset-ui-core/src/color/utils.ts:64]: I appreciate the explanation of this interleaving. This is one of those little bits of code where there are 90 ways to do it. We might be able to squeeze some performance here, but I don't think anyone's palette will be big enough to cause a perceivable performance problem anway.

**Review by rusackas — APPROVED** [review]
LGTM! Seems pretty clean... left one question on the thread about the `3` value that has me a little confused. I'll also spin up an environment to test it further for anyone interested, e.g. @jinghua-qa or @geido :)

File: superset-frontend/packages/superset-ui-core/src/color/utils.ts (hunk 1 above)
Comment (stephenLYZ) [reply superset-frontend/packages/superset-ui-core/src/color/utils.ts:58]: Good catch! This is to solve the problem that the first three values generated by `tinycolor.analogous` may have the same or very close colors. You can see my local test:
image
Let me add some comments here.

**Review by geido — APPROVED** [review]
LGTM!

**Comment by sadpandajoe** [comment]
🏷️ preset:2022.13

**Pull request merged**